Mage.Sets/src/mage/cards/s/SixSidedDie.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * @author Sidorovich77
+ */
+public final class SixSidedDie extends CardImpl {
+
+    public SixSidedDie(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.INSTANT}, "{2}{B}");
+
+        // Choose target creature. Roll a six-sided die.
+        //
+        //1 — It has base toughness 1 until end of turn.
+        //
+        //2 — Put two -1/-1 counters on it.
+        //
+        //3 — Six-Sided Die deals 3 damage to it, and you gain 3 life.
+        //
+        //4 — It gets -4/-4 until end of turn.
+        //
+        //5 — Destroy it.
+        //
+        //6 — Exile it.
+        this.getSpellAbility().addTarget(new TargetCreaturePermanent());
+        this.getSpellAbility().addEffect(new SixSidedDieEffect());
+    }
+
+    private SixSidedDie(final SixSidedDie card) {
+        super(card);
+    }
+
+    @Override
+    public SixSidedDie copy() {
+        return new SixSidedDie(this);
+    }
+}
+
+class SixSidedDieEffect extends OneShotEffect {
+
+    SixSidedDieEffect() {
+        super(Outcome.Benefit);
+        setText("choose target creature. Roll a six-sided die." +
+                "<br>1 — It has base toughness 1 until end of turn." +
+                "<br>2 — Put two -1/-1 counters on it." +
+                "<br>3 — Six-Sided Die deals 3 damage to it and you gain 3 life." +
+                "<br>4 — It gets -4/-4 until end of turn." +
+                "<br>5 — Destroy it." +
+                "<br>6 — Exile it.");
+    }
+
+    private SixSidedDieEffect(final SixSidedDieEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public SixSidedDieEffect copy() {
+        return new SixSidedDieEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player != null) {
+            int result = player.rollDice(outcome, source, game, 6);
+            Effect effect = null;
+            if (result == 1) {
+                game.addEffect(new SetBasePowerToughnessTargetEffect(1, 1, Duration.EndOfTurn), source);
+                return true;
+            } else if (result == 2) {
+                effect = new AddCountersTargetEffect(CounterType.M1M1.createInstance(2));
+            } else if (result == 3) {
+                effect = new DamageTargetEffect(3);
+                Player you = game.getPlayer(source.getControllerId());
+                if (you != null) {
+                    player.gainLife(3, game, source);
+                }
+            } else if (result == 4) {
+                game.addEffect(new BoostTargetEffect(-4, -4, Duration.EndOfTurn), source);
+                return true;
+            } else if (result == 5) {
+                effect = new DestroyTargetEffect();
+            } else if (result == 6) {
+                effect = new ExileTargetEffect();
+            }
+
+            if (effect != null) {
+                effect.apply(game, source);
```

Comment:
instead of creating the one shot effects and applying them here, just do whatever thing you need to do in that particular case of the switch statement